kube/pod_test.go

```diff
@@ -271,18 +436,766 @@ func TestPodGetEnvVars(t *testing.T) {
+			Name: "KUBE_SIZING_FOO_PORTS_STORE_MIN",
+		},
+		&model.ConfigurationVariable{
+			Name: "KUBE_SIZING_FOO_PORTS_STORE_MAX",
+		},
+	}, ExportSettings{
+		RoleManifest: &model.RoleManifest{
+			Roles: []*model.Role{
+				&model.Role{
+					Name: "foo",
+					Run: &model.RoleRun{
+						ExposedPorts: []*model.RoleRunExposedPort{
+							&model.RoleRunExposedPort{
+								Name:                "store",
+								CountIsConfigurable: true,
+								InternalPort:        333,
+								Count:               55,
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+
+	evYAML, err := testhelpers.RenderNode(ev, nil)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+- name: "KUBERNETES_NAMESPACE"
+  valueFrom:
+    fieldRef:
+      fieldPath: "metadata.namespace"
+- name: "KUBE_SIZING_FOO_PORTS_STORE_MAX"
+  value: "387"
+- name: "KUBE_SIZING_FOO_PORTS_STORE_MIN"
+  value: "333"
+`
+	assert.Equal(expectedYAML, string(evYAML))
+}
+
+func TestPodGetEnvVarsFromConfigSizingPortsHelm(t *testing.T) {
+	assert := assert.New(t)
+
+	ev, err := getEnvVarsFromConfigs([]*model.ConfigurationVariable{
+		&model.ConfigurationVariable{
+			Name: "KUBE_SIZING_FOO_PORTS_STORE_MIN",
+		},
+		&model.ConfigurationVariable{
+			Name: "KUBE_SIZING_FOO_PORTS_STORE_MAX",
+		},
+	}, ExportSettings{
+		CreateHelmChart: true,
+		RoleManifest: &model.RoleManifest{
+			Roles: []*model.Role{
+				&model.Role{
+					Name: "foo",
+					Run: &model.RoleRun{
+						ExposedPorts: []*model.RoleRunExposedPort{
+							&model.RoleRunExposedPort{
+								Name:                "store",
+								CountIsConfigurable: true,
+								InternalPort:        333,
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+
+	config := map[string]interface{}{
+		"Values.sizing.foo.ports.store.count": "22",
+	}
+
+	evYAML, err := testhelpers.RenderNode(ev, config)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+- name: "KUBERNETES_NAMESPACE"
+  valueFrom:
+    fieldRef:
+      fieldPath: "metadata.namespace"
+- name: "KUBE_SIZING_FOO_PORTS_STORE_MAX"
+  value: "354"
+- name: "KUBE_SIZING_FOO_PORTS_STORE_MIN"
+  value: "333"
+`
+	assert.Equal(expectedYAML, string(evYAML))
+}
+
+func TestPodGetEnvVarsFromConfigGenerationCounterKube(t *testing.T) {
+	assert := assert.New(t)
+
+	ev, err := getEnvVarsFromConfigs([]*model.ConfigurationVariable{
+		&model.ConfigurationVariable{
+			Name: "KUBE_SECRETS_GENERATION_COUNTER",
+		},
+	}, ExportSettings{
+		RoleManifest: &model.RoleManifest{
+			Roles: []*model.Role{
+				&model.Role{
+					Name: "foo",
+				},
+			},
+		},
+	})
+
+	evYAML, err := testhelpers.RenderNode(ev, nil)
+	if !assert.NoError(err) {
+		return
+	}
+
+	expectedYAML := `---
+- name: "KUBERNETES_NAMESPACE"
+  valueFrom:
+    fieldRef:
+      fieldPath: "metadata.namespace"
+- name: "KUBE_SECRETS_GENERATION_COUNTER"
+  value: "1"
+`
+	assert.Equal(expectedYAML, string(evYAML))
+}
+
+func TestPodGetEnvVarsFromConfigGenerationCounterHelm(t *testing.T) {
```

Comment:
All of these should probably be sub-tests of one thing

Reply:
Will check